include/RevProc.h

```diff
@@ -122,61 +124,38 @@ class RevProc{
 
   RevMem& GetMem() const { return *mem; }
 
-  /// RevProc: Add a RevThreadCtx to the Proc's ThreadTable
-  bool AddCtx(RevThreadCtx& Ctx);
-
-  /// RevProc: Create a new RevThreadCtx w/ Parent is currently executing thread
-  uint32_t CreateChildCtx();
-
-  /// RevProc: Get the ThreadState of a thread (pid) from the ThreadTable
-  // ThreadState GetThreadState(uint32_t pid) const;
-
-  /// RevProc: Set the ThreadState of a thread (pid) from the ThreadTable
-  // bool SetState(ThreadState, uint32_t pid);
+  std::bitset<_REV_HART_COUNT_>& GetThreadStateChanges(){ return ThreadStateChanges; }
 
-  /// RevProc: Used to pause RevThreadCtx w/ PID = pid
-  // bool PauseThread(uint32_t pid);
-
-  /// RevProc: Used to ready RevThreadCtx w/ PID = pid
-  // bool ReadyThread(uint32_t pid);
+  /// RevProc: SpawnThread creates a new thread and returns its ThreadID
+  void CreateThread(uint32_t NewTid, uint64_t fn, void* arg);
 
   /// RevProc: Returns the current HartToExec active pid
-  uint32_t GetActivePID() const { return GetActivePID(HartToExec); }
-
-  /// RevProc: Returns the active pid for HartID
-  uint32_t GetActivePID(const uint32_t HartID) const { return ActivePIDs.at(HartID); }
+  uint32_t GetActiveThreadID();
 
-  /// RevProc: Returns full list of PIDs (keys in ThreadTable)
-  std::vector<uint32_t> GetPIDs();
+  /// RevProc: Queue of threads to be spawned (RevProc creates the thread, RevCPU readies it for execution)
+  std::queue<std::shared_ptr<RevThread>>& GetNewThreadInfo() { return NewThreadInfo; }
 
   /// RevProc: Retires currently executing thread & then swaps to its parent. If no parent terminates program
   uint32_t RetireAndSwap(); // Returns new pid
 
-  /// RevProc: Used to raise an exception indicating a thread switch is coming (NewPID = PID of Ctx to switch to)
-  void CtxSwitchAlert(uint32_t NewPID) { NextPID=NewPID;PendingCtxSwitch = true; }
-
-  uint32_t HartToExecPID();
-  uint32_t HartToDecodePID();
-
-  ///< RevProc: Returns pointer to current ctx loaded into HartToExec
-  std::shared_ptr<RevThreadCtx> HartToExecCtx();
-
-  ///< RevProc: Returns pointer to current ctx loaded into HartToDecode
-  std::shared_ptr<RevThreadCtx> HartToDecodeCtx();
-
-  ///< RevProc: Change HartToExec active pid
-  bool ChangeActivePID(uint32_t PID);
-
   ///< RevProc: Change HartID active pid
-  bool ChangeActivePID(uint32_t PID, uint16_t HartID);
+  bool ChangeActiveThreadID(uint32_t ThreadID, uint16_t HartID);
+
+  ///< RevProc: Holds the info for all new threads to be spawned
+  ///           Right now this is through the following functions:
+  ///           - rev_pthread_create
+  std::queue<std::shared_ptr<RevThread>> NewThreadInfo;
 
-  bool UpdateRegFile();
-  // bool UpdateRegFile(uint16_t HartID);
+  ///< RevProc: Used for scheduling in RevCPU (if Utilization < 1, there is at least 1 unoccupied HART )
+  double GetUtilization(){ return ((double)AssignedThreads.size() / _REV_HART_COUNT_) * 100; }
```

Comment:
More concise:
```c++
return AssignedThreads.size() * 100.0 / _REV_HART_COUNT_;
```
By putting `* 100.0` in the middle, you force all operands to be converted to `double` without an explicit cast.

Reply:
Do you have any suggestions for a rename? HartUtilization perhaps?

Reply:
HartUtilization works for me